flask_bootstrap/templates/bootstrap/table.html

```diff
@@ -1,10 +1,30 @@
-{% from 'bootstrap/utils.html' import render_icon %}
+{% from 'bootstrap/utils.html' import render_icon, arg_url_for %}
 
 {% macro deprecate_old_pk_placeholder() %}
     {{ warn('The default action primary key placeholder has changed to ":id", please update.
         The support to the old value (":primary_key") will be removed in version 2.0.') }}
 {% endmacro %}
 
+{% macro build_url(endpoint, query, pk, kwargs) %}
+    {% with url_params = {} -%}
+        {%- do url_params.update(request.view_args if not endpoint else {}),
+            url_params.update(request.args if not endpoint else {}) -%}
+        {% with record = query.get(pk) %}
```

Comment:
Based on the usage here, there seems no difference between passing a Model and a Query.

Reply:
Agreed. I made the switch from `model` to `query` based on @michaelbukachi 's use-case

Reply:
The reasoning for changing the verbiage is to indicate that a filtered query can be supplied in addition to an all-encompassing query on the model object. So, if a user wants to pass in a model they would in turn use `query=ModelName.query`. Whereas, if they wish to provide a filtered query they could also do `query=ModelName.query.filter_by(active=True)`. But, in reality it doesn't really make a difference IMO as well. If you'd like me to revert it to model I can do that. Otherwise, I see no harm in leaving it as-is. Just let me know!

Reply:
I would prefer to use the `model` object here. Since we only need to call `get()` on the passed model instead of display the result, there is no need to support passing a `query` object.

Reply:
@greyli this means the table macro will be unusable for scenarios where not all results from a table required. Which is a really frequent use case 🤔

Reply:
I'll cede to you two deciding which direction we should go

Reply:
> this means the table macro will be unusable for scenarios where not all results from a table required.

@michaelbukachi As far as I can see, we only use this model/query argument to get the current record when building action URLs:

```
{% with record = query.get(pk) %}
```

So there is no difference between query and model.

Could you provide some explanation about the scenarios where we need to use a query object here?

Reply:
@greyli Oh, I was referring to the `query` parameter for the table macro. I mixed up conversations when replying using the GitHub app. How silly of me.

Reply:
Haha, well, so we can change back to `model` now.

Reply:
@caffeinatedMike It would be great if you can make the change. Sorry for the inconvenience :sweat_smile:.

Reply:
No worries @michaelbukachi ! Changes pushed @greyli 😃